src/component/Greeting.js

```diff
@@ -0,0 +1,23 @@
+import React, { useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { fetchGreetingApi } from '../redux/greetings/greetings';
+
+const Greeting = () => {
+  const greeting = useSelector((state) => state.greetingsReducer);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchGreetingApi());
+  }, [dispatch]);
```

Comment:
- [ ] <b>OPTIONAL:</b> This is not good for the app performance. look, you are using `useEffect` with dependency array `[dispatch]`. 
```js
useEffect(() => {
    dispatch(fetchGreetingApi());
}, [dispatch]);
```
This basically means that every time the dispatch changes, `useEffect` will run that function in its first parameter.
and for `useEffect` to know that that `dispatch` has changed,  it will compare the previous one and the new one by strict equal  `===`.  and of course, it will always return false since functions, as well as objects or arrays, are not primitive values.  which means two functions do not have referential equality. so `oneFunction === anotherFunction` will always return false. And that is what will happen to that `useEffect`, Therefore that condition is useless.  you can use [useCallback](https://reactjs.org/docs/hooks-reference.html#usecallback) to fix that, but either way that's not really the point. Here we just want to fetch a greeting first, when the component is mounted, so we don't need to pass anything in  useEffect's dependency array. 
```js
useEffect(() => {
    dispatch(fetchGreetingApi());
}, []);
```